.github/workflows/ci.yml

```diff
@@ -0,0 +1,30 @@
+name: CI
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
```

Comment:
Very cool! Two little questions:

- I don't know much about GitHub CI, but if the action (and with it the deployment of gh-pages) is run on pull requests, doesn't that mean that the API documentation is updated even for unmerged PRs?
- Could it make more sense to only run this action when a new tag/release is created? I'm undecided on this.

Reply:
You are right.
I've now split into 2 workflows.
The first on builds the docs on every push, pr (for testing).
The second runs if a new tag is created and pushes to gh-pages.